src/ParallelAlgorithms/Amr/Criteria/Random.hpp

```diff
@@ -16,49 +18,48 @@
 #include "Utilities/TMPL.hpp"
 
 namespace amr::Criteria {
+namespace detail {
+amr::Flag random_flag(
+    const std::unordered_map<amr::Flag, size_t>& probability_weights);
+}  // namespace detail
+
 /*!
- * \brief Randomly h-refine (or coarsen) an Element in each dimension.
- *
- * \details Let \f$f\f$ be `ChangeRefinementFraction`, \f$L_{max}\f$ be
- * `MaximumRefinementLevel`, and \f$L_d\f$ be the current refinement level
- * of an Element in a particular dimension.  In each dimension, a random
- * number \f$r_d \in [0, 1]\f$ is generated.  If \f$r_d > f\f$ the refinement
- * flag is set to amr::Flags::DoNothing.  If \f$r_d < f L_d / L_{max}\f$
- * the refinement flag is set to amr::Flags::Join.  Otherwise the
- * refinement flag is set to amr::Flag::Split.
+ * \brief Randomly refine (or coarsen) an Element in each dimension.
  *
- * \note This criterion is primarily useful for testing the mechanics of
- * h-refinement
+ * You can specify a probability for each possible `amr::Flag`. It is evaluated
+ * in each dimension separately.
```

Comment:
mention here (or the option help) that:
- probability for a flag is its weight over sum of weights
- you do not need to specify flags with weight 0
- if all weights are 0, DoNothing is always chosen